plugins/compress/misc.cc

```diff
@@ -41,40 +42,69 @@ gzip_free(voidpf /* opaque ATS_UNUSED */, voidpf address)
   TSfree(address);
 }
 
+namespace
+{
+// Strips parameters from value.  Returns cleared TextView if a q=f parameter present, where f is less than or equal to
+// zero.
+//
+void
+strip_ae_value(ts::TextView &value)
+{
+  ts::TextView compression{value.take_prefix_at(';')};
+  compression.trim(" \t");
+  while (value) {
+    ts::TextView param{value.take_prefix_at(';')};
+    ts::TextView name{param.take_prefix_at('=')};
+    name.trim(" \t");
+    if (strcasecmp("q", name) == 0) {
+      // If q value is valid and is zero, suppress compression types.
+      param.trim(" \t");
+      if (param) {
+        ts::TextView whole{param.take_prefix_at('.')};
+        whole.ltrim(" \t");
+        if ("0" == whole) {
```

Comment:
So `q=0000`0 is would remove the encoding, but `q=0.0` wouldn't?

The code could be simplified by just looking at `q=0`.

Reply:
This is what the spec says about what is a valid q parameter:  https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.9

What I think I see in core TS is that, if the q parameter is malformed, act as thought it was not present at all:
https://github.com/apache/trafficserver/blob/c40271ae7543a87bc90b4278954b79d304bb70b0/proxy/http/HttpTransactCache.cc#L768
https://github.com/apache/trafficserver/blob/c40271ae7543a87bc90b4278954b79d304bb70b0/proxy/hdrs/HttpCompat.cc#L479

The core code is more permissive than mine, but that doesn't necessarily seem good.  My code is safer and slightly more performant I think.